homeassistant/components/zwave_js/climate.py

```diff
@@ -479,3 +483,34 @@ async def async_set_preset_mode(self, preset_mode: str) -> None:
         if preset_mode_value is None:
             raise ValueError(f"Received an invalid preset mode: {preset_mode}")
         await self.info.node.async_set_value(self._current_mode, preset_mode_value)
+
+
+class DynamicCurrentTempClimate(ZWaveClimate):
+    """Representation of a thermostat that can dynamically use a different Zwave Value for current temp."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, client: ZwaveClient, info: ZwaveDiscoveryInfo
+    ) -> None:
+        """Initialize thermostat."""
+        super().__init__(config_entry, client, info)
+        if not self.info.platform_data:
+            raise ValueError("Expected platform data is missing from discovery info")
+
+        self._lookup_table: dict[
+            str | int, ZwaveValue | None
+        ] = self.info.platform_data["lookup_table"]
+        self._dependent_value: ZwaveValue | None = self.info.platform_data[
+            "dependent_value"
+        ]
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if self._dependent_value:
+            lookup_key = self._dependent_value.metadata.states[
```

Comment:
Can we still put the data template instance accessible on `self.info` and have the data template have a function that takes the resolved data and returns the zwave value? That way we keep the knowledge about how to parse the dependent value in the data template that generated it.

Reply:
done!